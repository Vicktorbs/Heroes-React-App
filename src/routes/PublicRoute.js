import React from 'react'
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router-dom'

export const PublicRoute = ({
    isNotAuthenticated,
    component: Component,
    ...rest
}) => {
    return (
        <Route { ...rest }
            component={ (props) => (
                (isNotAuthenticated) ?
                    (<Component { ...props } />) :
                    (<Redirect to='/' />)
            )}
        />
    )
}

PublicRoute.protoTypes = {
    isNotAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}
