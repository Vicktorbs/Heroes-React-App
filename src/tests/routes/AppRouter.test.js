import React from 'react';
import { mount } from "enzyme"
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routes/AppRouter';

describe('Test on router <AppRouter />', () => {
    
    const contexValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }
    test('should show login if it is not authenticated', () => {
        const wrapper = mount(
            <AuthContext.Provider value={ contexValue }>
                <AppRouter />
            </AuthContext.Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show DC component if it is authenticated', () => {
        const contexValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'VÃ­ctor'
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={ contexValue }>
                <AppRouter />
            </AuthContext.Provider>
        );
        
        expect(wrapper.find('.navbar').exists()).toBeTruthy();
    })
    
})
