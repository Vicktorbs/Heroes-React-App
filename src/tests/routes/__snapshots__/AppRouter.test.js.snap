// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on router <AppRouter /> should show login if it is not authenticated 1`] = `
<div>
  <div
    className="container mt-4"
  >
    <h3>
      Login!
    </h3>
    <hr />
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
